x-pack/plugins/apm/server/routes/fleet/source_maps.ts

```diff
@@ -54,11 +55,32 @@ export async function listArtifacts({
   fleetPluginStart: FleetPluginStart;
 }) {
   const apmArtifactClient = getApmArtifactClient(fleetPluginStart);
-  const artifacts = await apmArtifactClient.listArtifacts({
+
+  const artifacts = [];
+  const perPage = 100;
+  let page = 1;
+
+  let fleetArtifactsResponse = await apmArtifactClient.listArtifacts({
     kuery: 'type: sourcemap',
+    perPage,
+    page,
   });
+  artifacts.push(...fleetArtifactsResponse.items);
+
+  while (
+    fleetArtifactsResponse.total > artifacts.length &&
+    !isEmpty(fleetArtifactsResponse.items)
+  ) {
+    page += 1;
+    fleetArtifactsResponse = await apmArtifactClient.listArtifacts({
+      kuery: 'type: sourcemap',
+      perPage,
+      page,
+    });
+    artifacts.push(...fleetArtifactsResponse.items);
+  }
```

Comment:
I'm wondering if this would be more readable as a recursive function or using reduce.

Reply:
It was mostly the repetition of `apmArtifactClient.listArtifacts` that felt off. But I think you can keep it as-is unless you can think of a better solution.